File: app/src/main/java/java/android/notes/wrapper/fragments/extra/SettingsFragment.java
package java.android.notes.wrapper.fragments.extra;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import java.android.notes.R;
import java.android.notes.wrapper.helpers.CreateFragment;

public class SettingsFragment extends Fragment implements View.OnClickListener{

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        setHasOptionsMenu(true);
        return inflater.inflate(R.layout.fragment_settings,container,false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        view.findViewById(R.id.buttonEditDate).setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()){
            case R.id.buttonEditDate:
                CreateFragment.createCalendarFragment((AppCompatActivity) requireActivity());
                break;
        }
    }
}

File: app/src/main/java/java/android/notes/wrapper/fragments/extra/SettingsFragment.kt
package java.android.notes.wrapper.fragments.extra

import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import java.android.notes.R
import java.android.notes.wrapper.helpers.CreateFragment
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment

class SettingsFragment : Fragment(), View.OnClickListener {
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        setHasOptionsMenu(true)
        return inflater.inflate(R.layout.fragment_settings, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        view.findViewById<View>(R.id.buttonEditDate).setOnClickListener(this)
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.buttonEditDate -> CreateFragment.createCalendarFragment(requireActivity() as AppCompatActivity)
        }
    }
}